src/Octoshift/Services/GithubApi.cs

```diff
@@ -301,7 +301,7 @@ public virtual async Task<string> CreateGhecMigrationSource(string orgId)
         }
     }
 
-    public virtual async Task<string> StartMigration(string migrationSourceId, string sourceRepoUrl, string orgId, string repo, string sourceToken, string targetToken, string gitArchiveUrl = null, string metadataArchiveUrl = null, bool skipReleases = false, bool lockSource = false)
+    public virtual async Task<string> StartMigration(string migrationSourceId, string sourceRepoUrl, string orgId, string repo, string sourceToken, string targetToken, string gitArchiveUrl = null, string metadataArchiveUrl = null, bool skipReleases = false, string targetRepoVisibility = null, bool lockSource = false)
```

Comment:
I definitely recommend using an `enum` for the repo visibility, it's more strongly types and less error prone:
```c#
enum RepoVisiblity
{
    Private,
    Public,
    Internal
}
```

Reply:
I'm not sure that's necessary given the validation done by the command line library with this line:
```
AddOption(TargetRepoVisibility.FromAmong("public", "private", "internal"));
```

Reply:
This is the in the app level but it's also a good idea to do it in the API level too. IMO in general it's better API design and makes it easier to follow the logic. Despite the fact where it's gonna get used.

Reply:
This is the in the app level but it's also a good idea to do it in the API level too. IMO in general it's better API design and makes it easier to follow the logic. Despite the fact where it's gonna get used.